wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -400,6 +410,20 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
 
           for {
             _ <- insertIncomingTransaction(transaction, totalIncoming)
+            prevTagDbs <- addressTagDAO.findTx(transaction.txIdBE,
+                                               networkParameters)
+            prevTags = prevTagDbs.map(tagDb =>
+              AddressTag(tagDb.tag.tag, Some(tagDb.tagType)))
+            tagsToUse = prevTags
+              .filterNot(tag => newTags.contains(tag)) ++ newTags
+            newTagDbs = relevantOutsWithIdx.flatMap { out =>
+              val address = BitcoinAddress
+                .fromScriptPubKey(out.output.scriptPubKey, networkParameters)
+                .get
```

Comment:
Why `.get`? Shouldn't we just fail the future if we fail to parse the bitcoin address?

Reply:
It should be safe to assume that every one can parse to a bitcoin address, it can only fail if it is not a `P2PKH`, `P2SH`, or `WitnessSPK`. Since our wallet can only receive funds from these we should never be spending from something else.

Reply:
I don't think we ever plan on adding spending from `RawSPK`s as well

Reply:
I think it is a cleaner way to propogate the future by wrapping it inside of `Future.failed()` and returning that rather than just possibly throwing the exception. I believe there is a `Future.fromTry()` helper method exactly for this.

Reply:
It will still propagate to a failed future because it is within the `for`

Reply:
The error messsage isn't retained when calling `.get`, compared to using `Future.fromTry()`. When debugging you will get a `NoSuchElementException` rather than a more descriptive error message.